rts/motoko-rts/src/allocator.rs

```diff
@@ -0,0 +1,43 @@
+// c.f. https://os.phil-opp.com/heap-allocation/#dynamic-memory
+
+use alloc::alloc::{GlobalAlloc, Layout};
+//use core::ptr::null_mut;
+use crate::memory::{alloc_blob, ic};
+use crate::types::Bytes;
+
+pub struct HeapAllocator;
+
+unsafe impl GlobalAlloc for HeapAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let size = layout.size();
+        let align = layout.align();
+        // align is a power of 2
+        debug_assert!(align.count_ones() == 1);
+        if !(align <= 4) {
```

Comment:
Shouldn't this be `if align <= 4`? 
(or `if align <= WORD_SIZE as usize`)

Reply:
Oh yes! That was my intention indeed!

Reply:
At least I've tested the other path now too ;->